### PR TITLE
Update preprocessor doc #258

### PR DESCRIPTION
**Pull request by allieychen**
Update the preprocessor document.
- Add details on `infer headers`.
- Add one example of the report.

Issue: [258](https://github.com/googlegenomics/gcp-variant-transforms/issues/258).

### CODE REVIEW

File: docs/vcf_files_preprocessor.md [hunk 1]
```diff
@@ -20,9 +20,15 @@ This tool generates a report that includes three types of inconsistencies:
...
-  without a definition in the VCF files. The tool can infer the type for such
-  fields and provides them in the report.
+* Inferred headers (Optional): Occasionally, some fields may be used directly
+  without a definition in the VCF files, or though a definition is provided, the
+  field value does not match the field definition. This tool can infer the type
+  and num for such fields and provides them in the report. For the latter case,
+  two kinds of the mismatches are handled:
+  1. The defined field type is `Integer`, but the provided value is float.
+     Correct the type to be `Float`.
+  2. Defined num is `A`, but the provided values do not have the same
```
Comment (arostamianfar) [inline docs/vcf_files_preprocessor.md]: nit: please clarify what 'A' means (just add `(i.e. one value for each alternate base)`) as some people may not be familiar with the details of the VCF spec.

File: docs/vcf_files_preprocessor.md [hunk 2]
```diff
@@ -20,9 +20,15 @@ This tool generates a report that includes three types of inconsistencies:
...
+     cardinality as the alternate bases. Infer the num to be `None`.
```
Comment (arostamianfar) [inline docs/vcf_files_preprocessor.md]: nit: s/None/unknown (i.e. `.`)
(None is a PyVCF concept).

File: docs/vcf_files_preprocessor.md [hunk 3]
```diff
@@ -114,4 +120,31 @@ python -m gcp_variant_transforms.vcf_to_bq_preprocess \
...
+
+### Example
+
+The following is the conflicts report generated by running the preprocessor on
+the data set
+[1000 Genomes](https://cloud.google.com/genomics/docs/public-datasets/1000-genomes).
+
+Header Conflicts
+
+| ID  | Category |  Conflicts          |   File Paths |   Proposed Resolution |
```
Comment (arostamianfar) [inline docs/vcf_files_preprocessor.md]: Nice table!

File: docs/vcf_files_preprocessor.md (hunk 1 above)
Comment (allieychen) [reply docs/vcf_files_preprocessor.md]: Thanks, done.

File: docs/vcf_files_preprocessor.md (hunk 2 above)
Comment (allieychen) [reply docs/vcf_files_preprocessor.md]: Thank you for the info!

**Pull request merged**